SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/AppCenter.cs

```diff
@@ -114,22 +116,56 @@ static Task<bool> PlatformIsEnabledAsync()
             // It's fine for callers of the current implementation to use Wait().
             lock (AppCenterLock)
             {
-                return Task.FromResult(Instance.InstanceEnabled);
+                var taskSource = new TaskCompletionSource<bool>();
+                if (!Configured)
+                {
+                    AppCenterLog.Error(AppCenterLog.LogTag, NotConfiguredMessage);
+                    taskSource.SetException(new AppCenterException(NotConfiguredMessage));
+                }
+                else
+                {
+                    taskSource.SetResult(Instance.InstanceEnabled);
+                }
+                return taskSource.Task;
             }
         }
 
         static Task PlatformSetEnabledAsync(bool enabled)
         {
             lock (AppCenterLock)
             {
-                Instance.InstanceEnabled = enabled;
-                return Task.FromResult(default(object));
+                var taskSource = new TaskCompletionSource<object>();
+                if (!Configured)
+                {
+                    AppCenterLog.Error(AppCenterLog.LogTag, NotConfiguredMessage);
+                    taskSource.SetException(new AppCenterException(NotConfiguredMessage));
```

Comment:
our sdk never throws exception and should not start doing it as we don't want to crash prod apps unexpectedly, please use only error logs.

Also if it gives the right result before Configure we can leave the behavior, I think iOS works too. Android is different and might be improved.

Reply:
@guperrot Yeah, I'll revert this. Also, the most of the current problems with tests caused by this.